lg_earth/src/lg_earth/kmlsync.py

```diff
@@ -330,10 +330,24 @@ class KmlQueryHandler(tornado.web.RequestHandler):
     def initialize(self):
         self.playtour = self.application.playtour
         self.playtour_service = self.application.playtour_service
+        self.planet = self.application.planet
+        self.planet_service = self.application.planet_service
+        self.get_planet = self.application.get_planet
+
+    def _wait_for_planet(self, planet):
```

Comment:
On Mon, Feb 29, 2016 at 07:47:15AM -0800, Wojciech Ziniewicz wrote:
> Another possible problem - we're waiting in the loop for 5 seconds but is
> this wait blocking operation?

All I can say is that it's using rospy.sleep(). I admit I don't know just how
much that's going to block.

> If it's blocking (which means that **whole** ros node is waiting) then bad
> but acceptable in this case ( GE <3 ).
> 
> If it's non-blocking then kmlsync will accept and process incoming messages
> which may lead to random behavior.

Ideally it wouldn't block GE requesting updates, but would block more requests
to the query interface. That of course isn't really possible given that it all
uses the same tornado server on the same ROS node. The random behavior problem
you mention is definitely a problem in theory, but in practice we should
expect that sending lots of query messages in the same time frame isn't going
to work anyway. The queryfile interface is slow, and handles only one message
at a time, so, for example, we shouldn't ever have multiple assets within a
single scene that all expect to use the query interface, because they'll stomp
on each other. That's probably ok, because we'll really only want to play one
tour, or run one search, or execute a single flyto, at a time; the only time I
see us needing to issue more than one queryfile command in a scene is when we
need to switch planets before executing our search or starting our tour.

Actually, I wonder if we ought to modify the query node to serialize messages
intelligently rather than blatting them out to the query file as soon as
they're received...

> Can you please add some tests here?

Yeah, it definitely needs tests. I'm still learning how :)